services/authenticatation-service.js

```diff
@@ -0,0 +1,25 @@
+class AuthenticationService{
+
+    constructor(tokenService, userService){
+        this.tokenService = tokenService;
+        this.userService = userService;
+    }
+
+    async authenticate(req){
+        if (!req || !req.headers || !req.session || !req.session.userInfo) return false;
+        const token = getTokenFromRequest(req.headers['authorization']);
+        const UUIDArray = await this.userService.getUserTokenUUIDs(req.session.userInfo)
+        return Boolean((token && await this.tokenService.authenticateUserToken(token, UUIDArray)) || (!token && req.session.tokens))
+    }
+}
+
+function getTokenFromRequest(authHeader) {
+    if (!authHeader) return undefined;
```

Comment:
Is returning undefined necessary? I just dont want to return undefined or null. It seems this.tokenService.authenticateUserToken verifying undefined token.